cfpq_pyalgo/pygraphblas/grammars/boolean_matrix_rsm.py

```diff
@@ -0,0 +1,200 @@
+"""Base class for Recursive State Machine decomposed into Boolean Matrices"""
+
+from typing import Dict, List, Set, Tuple
+
+from pyformlang.finite_automaton import State, Symbol
+from pyformlang.rsa import Box, RecursiveAutomaton
+from pygraphblas import Matrix
+
+from cfpq_pyalgo.pygraphblas import BooleanMatrixGraph
+
+
+class BooleanMatrixRsm:
+    """A Recursive State Machine decomposed into Boolean Matrices class"""
+
+    def __init__(
+        self,
+        transitions: BooleanMatrixGraph,
+        labels: Set[str],
+        nonterminals: Set[str],
+        start_states: Dict[str, int],
+        final_states: Dict[str, List[int]],
+    ):
+        self._labels: Set[str] = labels
+        self._nonterminals: Set[str] = nonterminals
+        self._transitions: BooleanMatrixGraph = transitions
+        self._start_states: Dict[str, int] = start_states
+        self._final_states: Dict[str, List[int]] = final_states
+
+    def __getitem__(self, label: str) -> Matrix:
+        return self._transitions[label]
+
+    def __setitem__(self, label: str, matrix: Matrix):
+        self._transitions[label] = matrix
+
+    def __contains__(self, label: str) -> bool:
+        return label in self._transitions
+
+    @property
+    def matrices_size(self) -> int:
+        """The number of states in RSM
+
+        Returns
+        -------
+        matrices_size: int
+            Matrices size
+        """
+        return self._transitions.matrices_size
+
+    @property
+    def labels(self) -> Set[str]:
+        """All RSM labels"""
+        return self._labels
+
+    @property
+    def nonterminals(self) -> Set[str]:
+        """Labels of boxes"""
+        return self._nonterminals
+
+    def add_edge(self, u: int, v: int, label: str) -> None:
+        """Add an edge between `u` and `v` states with label `label.
+
+        The states `u` and `v` will be automatically added if they are
+        not already in the RSM.
+
+        Parameters
+        ----------
+        u: int
+            The tail of the edge
+
+        v: int
+            The head of the edge
+
+        label: str
+            The label of the edge
+        """
+        self._transitions.add_edge(u, v, label)
+
+    def get_start_state(self, label: str) -> int:
+        """Get start vertex for RSM box with label `label`
```

Comment:
Some doc strings with dot at the end, and some strings without dots. Please add dots everywhere.

Reply:
Fixed